src/app/services/actions/query-action-creator-util.ts

```diff
@@ -236,6 +236,7 @@ export function parseResponse(
         return response.json();
       case ContentType.XML:
       case ContentType.HTML:
+      case ContentType.TextCsv:
```

Comment:
@thewahome this looks good to me and works well.

I have a suggestion on handling these content types differently. Starting with the overview on media types https://www.rfc-editor.org/rfc/rfc2046.html#section-3, we can make a few changes on how we handle content types:
1. check the top-level media type of the content type in `getContentType`. This means, we return the type as `application`, `text`, or `image`. Alternativley, we can just split the content type on `/` and get the value.
2. we make the switch case to group the content types by type, therefore, all `text` types are rendered as text, `image` types can have their own image render and `application` we can process as we want i.e. json/xml -> format.
3. since we have few content types to parse, we can improve UI display by having customized previews of the response i.e. csv in a table.

NB: this is purely a good to have, as it is, it works well.

Reply:
I would love to explore this further. If you could have it as a feature request that would help us triage it better